var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -139,7 +139,10 @@ class Qt(Package):
     depends_on("libpng@1.2.57", when='@3')
     depends_on("pcre+multibyte", when='@5.0:5.8')
     depends_on("inputproto", when='@:5.8')
-    depends_on("openssl@:1.0.999", when='@:5.9+ssl')
+    if '+krellpatch':
```

Comment:
This syntax wrong work, a non-empty string always evaluates to True in Python

Reply:
Does this change look better?   It forces the old openssl when the krellpatch is not in effect.  It will only be in effect when we are building the qt3 version of qt for OpenSpeedShop

-    if '+krellpatch':
-        depends_on("openssl@:1.0.999", when='@4.0:5.9+ssl')
-    else:
-        depends_on("openssl@:1.0.999", when='@:5.9+ssl')
+    depends_on("openssl@:1.0.999", when='@:5.9+ssl~krellpatch')

Reply:
Sorry this should have been:
`diff --git a/var/spack/repos/builtin/packages/qt/package.py b/var/spack/repos/builtin/packages/qt/package.py
index 8b00959db..cbac9ed14 100644
--- a/var/spack/repos/builtin/packages/qt/package.py
+++ b/var/spack/repos/builtin/packages/qt/package.py
@@ -139,10 +139,7 @@ class Qt(Package):
     depends_on("libpng@1.2.57", when='@3')
     depends_on("pcre+multibyte", when='@5.0:5.8')
     depends_on("inputproto", when='@:5.8')
-    if '+krellpatch':
-        depends_on("openssl@:1.0.999", when='@4.0:5.9+ssl')
-    else:
-        depends_on("openssl@:1.0.999", when='@:5.9+ssl')
+    depends_on("openssl@:1.0.999", when='@:5.9+ssl~krellpatch')
 
     depends_on("glib", when='@4:')
     depends_on("libpng", when='@4:')
`

Reply:
If I replace this:

> -    if '+krellpatch':
> -        depends_on("openssl@:1.0.999", when='@4.0:5.9+ssl')
> -    else:
> -        depends_on("openssl@:1.0.999", when='@:5.9+ssl')

with this:   

> +    depends_on("openssl@:1.0.999", when='@:5.9+ssl~krellpatch')
 

it seems to work.   We will only use the old openssl when not building the qt3 gui for OpenSpeedShop.